addons/mail/static/tests/messaging/messaging.test.js

```diff
@@ -99,6 +110,8 @@ test("Show conversations with new message in chat hub (outside of discuss app)",
     );
     await click(".o-mail-DiscussSidebar-item:contains('Dumbledore'):has(.badge:contains(1))");
     await contains(".o-mail-Message:contains('Tricky')");
+    await expect.waitForSteps(["discuss.channel/new_message"]);
```

Comment:
I would expect it before the click on badge 1, this makes more sense new_message and badge to be asserted immediately next to each other (event before badge, since the event produces the badge).